lock/lock.py

```diff
@@ -80,59 +80,58 @@ async def lock(
-                    except:
-                        failed.append(inline(role.name))
-
-        msg = ""
-        if succeeded:
-            msg += f"{channel.mention} has been locked for {humanize_list(succeeded)}.\n"
-        if cancelled:
-            msg += f"{channel.mention} was already locked for {humanize_list(cancelled)}.\n"
-        if failed:
-            msg += f"I failed to lock {channel.mention} for {humanize_list(failed)}.\n"
-        if msg:
-            await ctx.send(msg)
+                    roles_or_members = roles_or_members[:10]
+                succeeded = []
+                cancelled = []
+                failed = []
+                reason = get_audit_reason(ctx.author)
+
+                if isinstance(channel, discord.TextChannel):
+                    for role in roles_or_members:
+                        current_perms = channel.overwrites_for(role)
+                        my_perms = channel.overwrites_for(ctx.me)
+                        if my_perms.send_messages != True:
+                            my_perms.update(send_messages=True)
+                            await channel.set_permissions(ctx.me, overwrite=my_perms)
+                        if current_perms.send_messages == False:
+                            cancelled.append(inline(role.name))
+                        else:
+                            current_perms.update(send_messages=False)
+                            try:
+                                await channel.set_permissions(role, overwrite=current_perms, reason=reason)
+                                succeeded.append(inline(role.name))
+                            except:
+                                failed.append(inline(role.name))
+                elif isinstance(channel, discord.VoiceChannel):
+                    for role in roles_or_members:
+                        current_perms = channel.overwrites_for(role)
+                        if current_perms.connect == False:
+                            cancelled.append(inline(role.name))
+                        else:
+                            current_perms.update(connect=False)
+                            try:
+                                await channel.set_permissions(role, overwrite=current_perms, reason=reason)
+                                succeeded.append(inline(role.name))
+                            except:
+                                failed.append(inline(role.name))
+
+                msg = ""
+                if succeeded:
+                    msg += f"{channel.mention} has been locked for {humanize_list(succeeded)}.\n"
+                if cancelled:
+                    msg += f"{channel.mention} was already locked for {humanize_list(cancelled)}.\n"
+                if failed:
+                    msg += f"I failed to lock {channel.mention} for {humanize_list(failed)}.\n"
+                if msg:
+                    await ctx.send(msg)
+        except discord.Forbidden:
+            return
```

Comment:
The try/except clause here absorbs all Forbidden errors. I think it's ok to just remove it.